src/middleware/indexer.js

```diff
@@ -136,21 +141,25 @@ const findLikelyTokens = async (ctx) => {
         where receipt_predecessor_account_id = $1
             and action_kind = 'FUNCTION_CALL'
             and (args->>'method_name' like 'ft_%' or args->>'method_name' = 'storage_deposit')
+            and receipt_included_in_block_timestamp <= $3
+            and receipt_included_in_block_timestamp > $4
     `;
 
     const ownershipChangeEvents = `
         select distinct emitted_by_contract_account_id as receiver_account_id 
         from assets__fungible_token_events
         where token_new_owner_account_id = $1
+            and emitted_at_block_timestamp <= $3
```

Comment:
`$3` is just the latest block timestamp, is there really any benefit in adding this?

Reply:
hmm we want to return the last block in the response so the client can query from that next time so I guess it's in case a block gets added during the queries? @frol can confirm but it's important we don't miss any blocks here since FTs / NFTs can be minted in those blocks.

Reply:
As @esaminu explained, this clause prevents potential race condition.